src/Services/ProxyAwareFactory.php

```diff
@@ -6,23 +6,32 @@
 
 class ProxyAwareFactory extends HubspotFactory
 {
-    /** @inheritDoc */
-    public function __construct($config = [], $client = null, $clientOptions = [], $wrapResponse = true)
-    {
-        if (null !== $client) {
-            $this->client = $client;
-        }
-
-        if ([] === $config['proxy']) {
-            parent::__construct($config, $client, $clientOptions, $wrapResponse);
-            return;
-        }
-
-        $this->client = new ProxyClient($config, null, $clientOptions, $wrapResponse);
-    }
-
-    public static function createClient(array $config = [], $client = null, $clientOptions = [], $wrapResponse = true)
-    {
-        return new static($config, $client, $clientOptions, $wrapResponse);
-    }
+	/** @inheritDoc */
+	public function __construct($config = [], $client = null, $clientOptions = [], $wrapResponse = true)
+	{
+		if (null !== $client)
+		{
+			$this->client = $client;
+		}
+
+		if (!array_key_exists('proxy', $config) || [] === $config['proxy'])
```

Comment:
```suggestion
		if ([] === $config['proxy'] || !array_key_exists('proxy', $config))
```
remember to put the most efficient and performant things in front. `===` is faster than iterating over an array

Reply:
agree, but first you should check if the key exists. Then try to get value from it, otherwise you will get notice

Reply:
oh that's right, my approach was strictly performance-based but You're right here